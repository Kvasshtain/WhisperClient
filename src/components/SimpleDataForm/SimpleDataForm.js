import React from 'react'
import PropTypes from 'prop-types'

import './SimpleDataForm.sass'

class SimpleDataForm extends React.Component {
  state = {
    newData: '',
  }

  onSubmit = event => {
    event.preventDefault()

    this.props.onSubmitNewData(this.state.newData)

    this.setState({
      newData: '',
    })
  }

  updateInputValue = eventArg => {
    this.setState({
      newData: eventArg.target.value,
    })
  }

  render() {
    const { caption, name, placeholder, type, buttonCaption } = this.props

    return (
      <form className="simpleDataForm" onSubmit={this.onSubmit}>
        <h4>{caption}</h4>
        <div>
          <input
            name={name}
            placeholder={placeholder}
            type={type}
            value={this.state.newData}
            onChange={this.updateInputValue}
          />
          <button type="submit">{buttonCaption}</button>
        </div>
      </form>
    )
  }
}

SimpleDataForm.propTypes = {
  onSubmitNewData: PropTypes.func.isRequired,
  caption: PropTypes.string,
  name: PropTypes.string.isRequired,
  placeholder: PropTypes.string,
  type: PropTypes.string.isRequired,
  buttonCaption: PropTypes.string,
}

export { SimpleDataForm }
