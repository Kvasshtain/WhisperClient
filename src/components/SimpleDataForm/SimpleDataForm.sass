@import "../ActiveFocusNoBorder.sass"
@import "../SimpleButton.sass"
    
$vertical-input-field-size: 30px
$font-color: #233527
$smallFont-size: 14px

$no-border: 0px solid white
    
.simpleDataForm
    padding: 10px
    background: linear-gradient(to bottom, #f5feea, #ace6ba)
    border-top: 1px solid #afafaf
    >h4
        color: $font-color
        margin: 0px
    >div
        display: flex
        >input
            height: $vertical-input-field-size
            width: 100%
            border-radius: $vertical-input-field-size 0 0 $vertical-input-field-size
            padding-left: 10px
            font-size: $smallFont-size
            border: $no-border
            background: #e1ffe7
            @include activeFocusNoBorder
        >button
            @include simpleButton
            height: $vertical-input-field-size + 2px
            padding-right: 25px
            padding-left: 20px
            border-radius: 0 $vertical-input-field-size $vertical-input-field-size 0
            border: $no-border
            font-size: $smallFont-size
            color: $font-color