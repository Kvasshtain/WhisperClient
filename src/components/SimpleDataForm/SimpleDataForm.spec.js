import React from 'react'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'

import { SimpleDataForm } from './SimpleDataForm'

describe('SimpleDataForm', () => {
  const mockOnSubmitNewData = jest.fn()

  it('should render correctly', () => {
    const testedComponent = shallow(
      <SimpleDataForm
        onSubmitNewData={mockOnSubmitNewData}
        u
        name="userMessage"
        type="text"
      />
    )

    expect(shallowToJson(testedComponent)).toMatchSnapshot()
  })
})

describe('when typing new message', () => {
  const mockOnSubmitNewData = jest.fn()
  const newTestData = 'Some new message'

  const testedComponent = shallow(
    <SimpleDataForm
      onSubmitNewData={mockOnSubmitNewData}
      name="userMessage"
      type="text"
    />
  )

  beforeEach(() => {
    testedComponent.find('input').simulate('change', {
      target: {
        value: newTestData,
      },
    })
  })

  it('updates inputValue field in state', () => {
    expect(testedComponent.state().newData).toEqual(newTestData)
  })
})

describe('when push the button', () => {
  const mockOnSubmitNewData = jest.fn()

  const myComponent = shallow(
    <SimpleDataForm
      onSubmitNewData={mockOnSubmitNewData}
      name="userMessage"
      type="text"
    />
  )

  beforeEach(() => {
    myComponent.find('form').simulate('submit', {
      preventDefault: () => {},
    })
  })

  it('calls the props.onSubmit', () => {
    expect(mockOnSubmitNewData).toHaveBeenCalledTimes(1)
  })
})
