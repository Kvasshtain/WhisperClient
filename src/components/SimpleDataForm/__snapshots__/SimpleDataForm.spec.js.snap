// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SimpleDataForm should render correctly 1`] = `
<form
  className="simpleDataForm"
  onSubmit={[Function]}
>
  <h4 />
  <input
    name="userMessage"
    onChange={[Function]}
    type="text"
    value=""
  />
  <button
    type="submit"
  />
</form>
`;
